_docs/tasks/security/per-service-mtls.md

```diff
@@ -0,0 +1,122 @@
+```
+
+```bash
+{
+  "origin": "127.0.0.1"
+}
+```
+
+```bash
+kubectl exec $(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name} -n legacy) -n legacy -- curl http://httpbin.default:8000/ip -s
+```
+
+```bash
+command terminated with exit code 56
+```
+
+## Disable mutual TLS authentication for "httpbin" service.
+
+If we want to disable mTLS only for httpbin (on port 8000), without changing the mesh authentication settings,
+we can do that by adding this annotations to the httpbin service definition.
+
+```bash
+annotations:
+  auth.istio.io/8000: NONE
+```
+
+For a quick test, run `kubectl edit svc httpbin` and add the annotations above (
+or you can edit the original httpbin.yaml file and re-apply it). After the change is applied, request from sleep.legacy should now success, as the result of mTLS was dropped.
+
+Note: the annotations can be used in the opposite direction, i.e enable mTLS for a services, simply by using annotation value  `MUTUAL_TLS`, instead of `NONE`.
+
+## Disable mutual TLS authentication for control services.
+
+As we cannot annotate control services, such as API server, in Istio 0.3, we introduced [mtls_excluded_services](https://github.com/istio/api/blob/master/mesh/v1alpha1/config.proto#L200:19) to the mesh configuration to specify the list of services for which mTLS should not be used. If your application needs to communicate to any control service, it's fully-qualified domain name should be listed there.
+
+In the part of the demo, we will show the impact of this field.
+
+By default (0.3 or later), this list contains `kubernetes.default.svc.cluster.local` (which is the name of the API server service in common setup). You can verify it by running this command:
+
+```bash
+kubectl get configmap -n istio-system istio -o yaml | grep mtlsExcludedServices
+```
+
+```bash
+mtlsExcludedServices: ["kubernetes.default.svc.cluster.local"]
+```
+
+It's then expected that request to kubernetes.default service should be possible:
+
+```bash
+kubectl exec $(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name}) -c sleep -- curl https://kubernetes.default:443/api/ -k -s
+```
+
+```bash
+{
+  "kind": "APIVersions",
+  "versions": [
+    "v1"
+  ],
+  "serverAddressByClientCIDRs": [
+    {
+      "clientCIDR": "0.0.0.0/0",
+      "serverAddress": "104.199.122.14"
+    }
+  ]
+}
+```
+
+Now, run `kubectl edit configmap istio -n istio-system` and clear mtlsExcludedServices and restart pilot after done:
+
+```bash
+kubectl get pod $(kubectl get pod -l istio=pilot -n istio-system -o jsonpath={.items..metadata.name}) -n istio-system -o yaml | kubectl replace --force -f -
+```
+
+The same test request above now fail with code 35, as sleep's sidecar starts using mTLS again:
+
+```bash
+kubectl exec $(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name}) -c sleep -- curl https://kubernetes.default:443/api/ -k -s
+```
+
+```bash
+command terminated with exit code 35
+```
```

Comment:
Should we add another section about the other direction communication, i.e., from control service to sidecar?

Reply:
If Istio service is as-server, then it falls in the first scenario, i.e we can disable TLS using annotation.

Reply:
After a second thought, what about rename the two sections to something like 

Disable mutual TLS for Envoy as a server
Disable mutual TLS for Envoy as a client

That will make them more clear. And maybe we can explicitly mention that health check belongs to the first case, as health check is something customers/developers care about a lot.

Reply:
Discussed offline, health check story is not complete yet. And we are waiting for multiplex port/alias port support.

Reply:
Recap from offline discussion: I tried to make this as a tutorial to use the new features (annotation, exclude mtls list) rather than a solution for a specific problem, as while it helps to walk around the issue, it's not the solution we want to provide at the end. We can have another doc in future to address health check solution for example.